pluginframework/resource_quality_monitor.go

```diff
@@ -0,0 +1,238 @@
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+)
+
+const qualityMonitorDefaultProvisionTimeout = 15 * time.Minute
+
+func WaitForMonitor(w *databricks.WorkspaceClient, ctx context.Context, monitorName string) error {
+	return retry.RetryContext(ctx, qualityMonitorDefaultProvisionTimeout, func() *retry.RetryError {
+		endpoint, err := w.QualityMonitors.GetByTableName(ctx, monitorName)
+		if err != nil {
+			return retry.NonRetryableError(err)
+		}
+
+		switch endpoint.Status {
+		case catalog.MonitorInfoStatusMonitorStatusActive:
+			return nil
+		case catalog.MonitorInfoStatusMonitorStatusError, catalog.MonitorInfoStatusMonitorStatusFailed:
+			return retry.NonRetryableError(fmt.Errorf("monitor status retrund %s for monitor: %s", endpoint.Status, monitorName))
+		}
+		return retry.RetryableError(fmt.Errorf("monitor %s is still pending", monitorName))
+	})
+}
+
+var _ resource.Resource = &QualityMonitorResource{}
+
+func ResourceQualityMonitor() func() resource.Resource {
+	return func() resource.Resource {
+		return &QualityMonitorResource{}
+	}
+}
+
+type MonitorInfoExtended struct {
+	catalog_tf.MonitorInfo
+	// Adding new fields.
+	WarehouseId          types.String `tfsdk:"warehouse_id" tf:"optional"`
+	SkipBuiltinDashboard types.Bool   `tfsdk:"skip_builtin_dashboard" tf:"optional"`
+}
+
+type QualityMonitorResource struct {
+	Client *common.DatabricksClient
+}
+
+func (r *QualityMonitorResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "databricks_lakehouse_monitor_pluginframework"
+}
+
+func (r *QualityMonitorResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Terraform schema for Databricks Lakehouse Monitor. MonitorInfo struct is used to create the schema",
+		Attributes: common.PluginFrameworkResourceStructToSchemaMap(MonitorInfoExtended{}, func(c common.CustomizableSchemaPluginFramework) common.CustomizableSchemaPluginFramework {
+			c.SetRequired("assets_dir")
+			c.SetRequired("output_schema_name")
+			c.SetRequired("table_name")
+			c.SetReadOnly("monitor_version")
+			c.SetReadOnly("drift_metrics_table_name")
+			c.SetReadOnly("profile_metrics_table_name")
+			c.SetReadOnly("status")
+			c.SetReadOnly("dashboard_id")
+			c.SetReadOnly("schedule", "pause_status")
+			return c
+		}),
+	}
+}
+
+func (d *QualityMonitorResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	client, ok := req.ProviderData.(*common.DatabricksClient)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *common.DatabricksClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
```

Comment:
Either make this common or have a constant diag.Diagnostics or func() diag.Diagnostics that can be used here.